presto-main/src/main/java/com/facebook/presto/operator/DynamicFilterStats.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator;
+
+import com.facebook.drift.annotations.ThriftConstructor;
+import com.facebook.drift.annotations.ThriftField;
+import com.facebook.drift.annotations.ThriftStruct;
+import com.facebook.presto.spi.plan.PlanNodeId;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.Set;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Records the dynamic filter execution stats for an operator on native worker.
+ */
+@ThriftStruct
+public class DynamicFilterStats
+{
+    private final Set<PlanNodeId> producerNodeIds;
```

Comment:
These PlanNodeIds will be the PlanNodeId of the joins which has dynamic filter pushdown?
How do we know which nodes these filters are pushed down to?

Reply:
DynamicFilterStats is set in operator stats of those nodes that has dynamic filters.

Reply:
For example, if we have a query plan like:
join <- (probe side) project <- filter <- scan
where the dynamic filter pushdown is enabled, we should have project, filter and scan all in the producerNodeIds, right?

Reply:
Synced offline, the plan node ids will be the where the dynamic filter is from, which is the join node in the example I showed above.